workloads/librispeech/librispeech_pytorch/models.py

```diff
@@ -0,0 +1,267 @@
+                1, 32, kernel_size=(41, 11), stride=(2, 2), padding=(20, 5)),
+            nn.BatchNorm2d(32), nn.Hardtanh(0, 20, inplace=True),
+            nn.Conv2d(
+                32, 32, kernel_size=(21, 11), stride=(2, 1), padding=(10, 5)),
+            nn.BatchNorm2d(32), nn.Hardtanh(0, 20, inplace=True)))
+
+    rnn_input_size = 161
+    rnn_input_size = int(math.floor(rnn_input_size + 2 * 20 - 41) / 2 + 1)
+    rnn_input_size = int(math.floor(rnn_input_size + 2 * 10 - 21) / 2 + 1)
+    rnn_input_size *= 32
+
+    rnns = []
+    rnn = BatchRNN(
+        input_size=rnn_input_size,
+        hidden_size=self.hidden_size,
+        rnn_type=supported_rnns[self.rnn_type],
+        bidirectional=self.bidirectional,
+        batch_norm=False)
+    rnns.append(("0", rnn))
+    for x in range(self.hidden_layers - 1):
+      rnn = BatchRNN(
+          input_size=self.hidden_size,
+          hidden_size=self.hidden_size,
+          rnn_type=supported_rnns[self.rnn_type],
+          bidirectional=self.bidirectional)
+      rnns.append(("%d" % (x + 1), rnn))
+    self.rnns = nn.Sequential(collections.OrderedDict(rnns))
+    self.lookahead = nn.Sequential(
+        Lookahead(self.hidden_size, context=self.context),
+        nn.Hardtanh(0, 20, inplace=True)) if not self.bidirectional else None
+
+    fully_connected = nn.Sequential(
+        nn.BatchNorm1d(self.hidden_size),
+        nn.Linear(self.hidden_size, self.num_classes, bias=False))
+    self.fc = nn.Sequential(SequenceWise(fully_connected),)
+
+    self.inference_softmax = InferenceBatchSoftmax()
+    self.pad_token = 0
+
+  def get_seq_lens(self, input_length):
+    """Get a 1D tensor or variable containing the size sequences that will be output by the network.
+
+    Args:
+      input_length: 1D Tensor
+
+    Returns:
+      1D Tensor scaled by model
+    """
+    seq_len = input_length
+    for m in self.conv.modules():
+      if isinstance(m, nn.modules.conv.Conv2d):
+        seq_len = torch.div(
+            seq_len + 2 * m.padding[1] - m.dilation[1] *
+            (m.kernel_size[1] - 1) - 1,
+            m.stride[1],
+            rounding_mode="trunc") + 1
+    return seq_len.int()
+
+  def forward(self, x, lengths, trns):
+    lengths = lengths.int()
+    output_lengths = self.get_seq_lens(lengths)
+    x, _ = self.conv(x, output_lengths)
+
+    sizes = x.size()
+    x = x.view(sizes[0], sizes[1] * sizes[2],
+               sizes[3])  # Collapse feature dimension
+    x = x.transpose(1, 2).transpose(0, 1).contiguous()  # TxNxH
+
+    for rnn in self.rnns:
+      x = rnn(x, output_lengths)
+
+    if not self.bidirectional:  # no need for lookahead layer in bidirectional
+      x = self.lookahead(x)
+
+    x = self.fc(x)
+    log_probs = x.log_softmax(dim=-1).transpose(0, 1)
+
+    return log_probs, output_lengths
+
+  def eval_loss(self, dataset, loss_fn, device):
+    self.eval()
+
+    total_loss = 0.0
+    total_count = 0.0
+    for (_, features, trns, input_lengths) in dataset:
```

Comment:
is `trns` a typical abbreviation in the speech field? would it be possible to make it a more explicit variable name?